test/jdk/jdk/security/jarsigner/JarWithOneNonDisabledDigestAlg.java

```diff
@@ -54,41 +54,53 @@
 public class JarWithOneNonDisabledDigestAlg {
 
     private static final String PASS = "changeit";
-    private static final String TESTFILE1 = "testfile1";
-    private static final String TESTFILE2 = "testfile2";
+    private static final Path TESTFILE1 = Path.of("testfile1");
+    private static final Path TESTFILE2 = Path.of("testfile2");
+    private static final Path UNSIGNED_JAR = Path.of("unsigned.jar");
+    private static final Path SIGNED_JAR = Path.of("signed.jar");
+    private static final Path SIGNED_TWICE_JAR = Path.of("signed2.jar");
+    private static final Path MULTI_SIGNED_JAR = Path.of("multi-signed.jar");
+    private static final Path CURRENT_DIR = Path.of(".");
 
     public static void main(String[] args) throws Exception {
         SecurityUtils.removeFromDisabledAlgs("jdk.jar.disabledAlgorithms",
-            List.of("SHA1"));
-        Files.write(Path.of(TESTFILE1), TESTFILE1.getBytes());
-        JarUtils.createJarFile(Path.of("unsigned.jar"), Path.of("."),
-            Path.of(TESTFILE1));
+                List.of("SHA256"));
```

Comment:
@wangweij 

I initially removed this code, then restored it because I thought the original author might have intended to future-proof the test. It also serves as a sort of documentation of the implicit assumtions the test makes about the permitted state of digest algorithms in the JVM.

I have now instead added a method which explicitly asserts that MD5 is disabled and SHA256 is permitted in the very beginning of the test. This way the assumtions are made clear and the test will fail clear and loudly should these assumtions fail in the future.

What do you think of this update?

Reply:
True, making assumptions clear is maybe more important than future-proofing here. 

(Although we do actually have at least one example of a signed jar test today where the jar is no longer treated as signed because the jar was signed with SHA-1 before 2019. See VerifySignedJar which seems to not actually test much)